frontend/packages/ceph-storage-plugin/src/components/ocs-install/node-list.tsx

```diff
@@ -311,20 +318,41 @@ const CustomNodeTable: React.FC<CustomNodeTableProps> = ({
 
     const ocsObj = _.cloneDeep(ocsRequestData);
     ocsObj.spec.storageDeviceSets[0].dataPVCTemplate.spec.storageClassName = storageClass;
-    promises.push(k8sCreate(OCSServiceModel, ocsObj));
-
-    Promise.all(promises)
-      .then(() => {
-        history.push(
-          `/k8s/ns/${ocsProps.namespace}/clusterserviceversions/${
-            ocsProps.clusterServiceVersion.metadata.name
-          }/${referenceForModel(OCSServiceModel)}/${ocsObj.metadata.name}`,
-        );
-      })
-      .catch((err) => {
-        setProgress(false);
-        setError(err.message);
-      });
+
+    if (promises) {
+      Promise.all(promises)
+        .then(() => {
+          k8sCreate(OCSServiceModel, ocsObj)
+            .then(() => {
+              history.push(
+                `/k8s/ns/${ocsProps.namespace}/clusterserviceversions/${
+                  ocsProps.clusterServiceVersion.metadata.name
+                }/${referenceForModel(OCSServiceModel)}/${ocsObj.metadata.name}`,
+              );
+            })
+            .catch((err) => {
+              setProgress(false);
+              setError(err.message);
+            });
+        })
+        .catch((err) => {
+          setProgress(false);
+          setError(err.message);
+        });
+    } else {
+      k8sCreate(OCSServiceModel, ocsObj)
+        .then(() => {
+          history.push(
+            `/k8s/ns/${ocsProps.namespace}/clusterserviceversions/${
+              ocsProps.clusterServiceVersion.metadata.name
+            }/${referenceForModel(OCSServiceModel)}/${ocsObj.metadata.name}`,
+          );
+        })
+        .catch((err) => {
+          setProgress(false);
+          setError(err.message);
+        });
+    }
```

Comment:
no need for `if else`. Use `finally`

Reply:
No need to use `finally` here either. [Promise.all()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) will resolve if passed an empty array. You can remove all of the conditional logic around this, and it should work fine.